File: student-backend/src/Logger/Logger.ts
require('dotenv').config();
const write_log = async(directory:string , file:string , line_number:number ,msg:string=" ")=>{
    const backend=process.env.BACKEND_ADDRESS;
    try {
        await fetch(`${backend}/log`,{
            method:'POST' , 
            headers:{'Content-Type' : 'application/json' } , 
            body: JSON.stringify({directory,file, line_number , message:msg})
        });
    } catch (error) {
        console.log("failed to send log : " + error);

    }
};

module.exports = { write_log };

File: Logging/Logger.py

from fastapi import FastAPI  
from pydantic import BaseModel

app=FastAPI()

class LogEntry(BaseModel):
    directory:str 
    file : str  
    line_number:int 
    message:str 


@app.post("/log")
async def write_log(entry:LogEntry):
    with open("log.txt","a") as f:
        f.write(f"directory: {entry.directory}  file: {entry.file}  line number: {entry.line_number} msg: {entry.message}")
    
    
